cmd/registry/controller/local-tasks.go

```diff
@@ -29,6 +29,24 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+// Implement io.Writer interface https://pkg.go.dev/io#Writer
+type LogWriter struct {
+	logger *log.Entry
+}
+
+func (w LogWriter) Write(p []byte) (n int, err error) {
+	// This is required to attach logger info to each line in multiline logs
+	for _, s := range strings.Split(strings.TrimRight(string(p), "\n"), "\n") {
+		w.logger.Debug(s)
```

Comment:
Do we know that this method is only called with complete lines? I'm concerned we might end up writing partial/incomplete lines with this approach.

Reply:
Using the logger on each call to `Write` modifies the output by splitting it into multiple log entries, so the same problem still exists without the newline splitting. I think we can either 1) write 3p command output to a buffer and print the entire buffer as one log entry, or 2) modify the LogWriter to buffer output until it sees a newline/EOF, writing the log only when a full line of output is received in the buffer.

If you've tested your existing solution and it doesn't have any issues, then I don't think we need to address this issue in this PR. It will be easy enough to follow up with a fix at a later time.

Reply:
I didn't find any issues with my current testing, but I agree that what you are pointing out is possible when we execute long running actions. I am leaning towards option 2) from your suggestion. Currently, we are waiting for the command to finish executing, however, eventually we would want to move away from that when we start supporting long running actions. I will create a bug to revisit this later. (https://github.com/apigee/registry/issues/310)